ceph_deploy/repo.py

```diff
@@ -0,0 +1,112 @@
+import logging
+
+from ceph_deploy import hosts
+from ceph_deploy.cliutil import priority
+
+
+LOG = logging.getLogger(__name__)
+
+
+def install_repo(distro, args, cd_conf, rlogger):
+    if args.repo_name in cd_conf.get_repos():
+        LOG.info('will use repository %s from ceph-deploy config', args.repo_name)
+        options = dict(cd_conf.items(args.repo_name))
+        extra_repos = cd_conf.get_list(args.repo_name, 'extra-repos')
+        try:
+            repo_url = options.pop('baseurl')
+            gpg_url = options.pop('gpgkey')
+        except KeyError as err:
+            raise RuntimeError(
+                'missing required key: %s in config section: %s' % (err, args.repo_name)
```

Comment:
Thank yourself!  It's a direct lift from: https://github.com/ceph/ceph-deploy/blob/master/ceph_deploy/install.py#L246

I'm hopeful to eventually be able to make the `install --repo` code use this `repo` code directly.